### PR TITLE
Arrays of Scalar Values

### PR DESCRIPTION
**Pull request by johnthepink**
Since lists from GraphQL should be of the same type, this basically just places the array in the store as is if the first non-null item in the array is a scalar.

Does this make sense, or do we want to try to denormalize it? That looks like it will be a little messier, as the GraphQL parser doesn't treat arrays of scalars that way.

### CODE REVIEW

**Comment by stubailo** [comment]
No I think doing it this way is great! We might want to think about nested arrays, but AFAIK people usually don't actually do that with GraphQL. Going to take a look now.

**Comment by stubailo** [comment]
Sigh #54 strikes again on Travis, restarted the build..

File: test/readFromStore.ts [hunk 1]
```diff
@@ -271,4 +271,70 @@ describe('reading from the store', () => {
...
+
+    // The result of the query shouldn't contain __data_id fields
+    assert.deepEqual(queryResult, {
+      stringField: 'This is a string!',
+      numberField: 5,
+      simpleArray: ['one', 'two', 'three'],
+    });
+  });
+
+  it('runs an array of non-objects with null', () => {
```
Comment (stubailo) [inline test/readFromStore.ts]: Should we add a test for an array of objects, some of which are null? Or is that a different PR for #65?

File: src/writeToStore.ts [hunk 2]
```diff
@@ -168,28 +170,42 @@ function writeFieldToStore({
...
+    });
 
-      if (! isString(clonedItem.id)) {
-        clonedItem['__data_id'] = `${dataId}.${storeFieldName}.${index}`;
-      } else {
-        clonedItem['__data_id'] = clonedItem.id;
-      }
+    // GraphQL lists should be of the same type.
+    // If it's an array of scalar values, don't normalize.
+    if (! isObject(firstNonNullValue)) {
```
Comment (stubailo) [inline src/writeToStore.ts]: I think a better way to do this is to check if `field.selectionSet` exists. Since the query:

```
{
  myArray
}
```

Necessarily returns an array of scalars, and

```
{
   myArray { a, b, c }
}
```

Is necessarily an array of objects.

File: test/readFromStore.ts (hunk 1 above)
Comment (johnthepink) [inline test/readFromStore.ts]: @stubailo yea I started adding tests for that case, and then it felt like a separate issue. I'm headed to #65 next

Comment (stubailo) [inline test/readFromStore.ts]: :+1:

File: src/writeToStore.ts (hunk 2 above)
Comment (johnthepink) [inline src/writeToStore.ts]: @stubailo good catch! I like that better, and I pushed the change.

**Comment by stubailo** [comment]
Looks great to me! Going to merge as soon as Travis builds.

**Comment by stubailo** [comment]
Frigging Coveralls... sigh

**Comment by stubailo** [comment]
Thanks!
